src/reducers/summit_speakers/summit-speakers-list-reducer.js

```diff
@@ -56,31 +58,74 @@ const summitSpeakersListReducer = (state = DEFAULT_STATE, action) => {
+                    orderDir,
+                    currentPage: page,
+                    ...rest
+                }
+            }
+
+            return {
+                ...state,
+                order,
+                orderDir,
+                currentPage: page,
+                selectedItems: [],
+                excludedItems: [],
+                selectedCount: 0,
+                selectedAll: false,
+                ...rest
+            }
         }
         case RECEIVE_SPEAKERS_BY_SUMMIT: {
             let { current_page, total, last_page } = payload.response;
 
+            const items = buildSpeakersSubmittersList(state, payload.response.data);
+
+
             return {
                 ...state,
-                items: buildSpeakersSubmittersList(state, payload.response.data),
+                items: markCheckedItems(items, state),
                 currentPage: current_page,
                 totalItems: total,
                 lastPage: last_page,
             };
         }
         case SELECT_SUMMIT_SPEAKER: {
-            return { ...state, selectedItems: [...state.selectedItems, payload], selectedAll: false }
+            let newState = {};
+
+            if (state.selectedAll) {
+                newState = { ...state, excludedItems: state.excludedItems.filter(it => it !== payload), selectedItems: [] }
+            } else {
+                newState = { ...state, selectedItems: [...state.selectedItems, payload], excludedItems: [] }
+            }
+
+            return {...newState, items: markCheckedItems(state.items, newState), selectedCount: state.selectedCount + 1}
         }
         case UNSELECT_SUMMIT_SPEAKER: {
-            return { ...state, selectedItems: state.selectedItems.filter(e => e !== payload), selectedAll: false }
+            let newState = {};
+
+            if (state.selectedAll) {
+                newState = { ...state, excludedItems: [...state.excludedItems, payload], selectedItems: [] }
+            } else {
+                newState = { ...state, selectedItems: state.selectedItems.filter(it => it !== payload), excludedItems: [] }
+            }
+
+            return {...newState, items: markCheckedItems(state.items, newState), selectedCount: state.selectedCount - 1}
         }
         case SELECT_ALL_SUMMIT_SPEAKERS: {
-            return { ...state, selectedAll: true, selectedItems:[] }
+            const newState = {...state, selectedAll: true, selectedItems: [], excludedItems: [] }
+            return { ...newState, items: markCheckedItems(state.items, newState), selectedCount: state.totalItems }
         }
         case UNSELECT_ALL_SUMMIT_SPEAKERS: {
-            return { ...state, selectedAll: false, selectedItems:[]  }
+            const newState = {...state, selectedAll: false, selectedItems: [], excludedItems: [] }
+            return { ...newState, items: markCheckedItems(state.items, newState), selectedCount: 0 }
         }
         case SEND_SPEAKERS_EMAILS: {
             return {
```

Comment:
here we need to add 

   selectedItems: [],
                excludedItems: [],
                selectedCount: 0,
                selectedAll: false,
 items: state.items.map(it => ({...it, checked: false}))